Scripts/Runtime/Core/AnimationSequencerController.cs

```diff
@@ -36,21 +47,38 @@ public enum PlayType
 
         [SerializeField]
         private UnityEvent onStartEvent = new UnityEvent();
-        public UnityEvent OnStartEvent => onStartEvent;
+        public UnityEvent OnStartEvent { get { return onStartEvent;} protected set {onStartEvent = value;}}
```

Comment:
In this case I believe would be better to just add a `AddOnStartEventCallbackListener(UnityAction callback)`; just to make sure someone cannot just erase all the callbacks by creating a new unity event using a setter by mistake.

Reply:
Hey Bruno, the setter is protected and I am using this to redirect OnStartEvent and OnFinsihedEvent based on the playback direction. It feels to me like the AddOnStartEventCallbackListene() etc would be different public functionality?